### ISSUE TITLE
API for Arena

### ISSUE THREAD

**Issue by TheYsconator** [rustyscreeps/screeps-game-api#356]
Are you guys planning on building the same toolset (the Rust bindings, the cargo deploy tools and an example project) for Screeps Arena as well?

If so, and if you would appreciate help with that; what would be some more easy tasks I could take on? I'm not that proficient in Rust yet, but I'm willing to learn and I do have time to work on something like this.

**Comment by shanemadden** [comment]
Yup, that's the plan! We've been able to get the library adapted to the new smaller API surface, and have successfully gotten test wasm modules loaded in the arena environment (albeit in a hacky way, encoded in base64 due to the new ES6 wasm import syntax needing extra work to integrate).

The last step for having a 'hello arena world' setup that I can share (and get help getting the API coverage completed) is getting the build tooling working, which I was able to get started on this morning - should have something usable soon!

**Comment by TheYsconator** [comment]
Ooohh that is really good to read! It sounds as though you are a lot further than I had expected, but still; don't hesitate to poke me if I can help with something.

Thanks for maintaining all of this.

**Comment by nboltn** [comment]
how's progress going on arena?

**Comment by shanemadden** [comment]
We're still pending getting the crate released, but closing this out as we've got dedicated repos for arena bindings that are mostly feature complete, and should have a crate published before too long 🎊 : https://github.com/rustyscreeps/screeps-arena-starter-rust and https://github.com/rustyscreeps/screeps-arena-game-api
